AWSAppSyncClient/AppSyncSubscriptionWithSync.swift

```diff
@@ -0,0 +1,466 @@
+        #else
+        timer.scheduleOneshot(deadline: deadline)
+        #endif
+        timer.setEventHandler(handler: block)
+        timer.resume()
+        return timer
+    }
+    
+    func setupAsyncPoll(deadline: DispatchTime) -> DispatchSourceTimer {
+        // Invalidate existing time and restart again
+        activeTimer?.cancel()
+        
+        return executeAfter(deadline: deadline, queue: self.handlerQueue!) {
+            AppSyncLog.debug("DS: Timer fired. Performing sync.")
+            self.deltaSyncOperationQueue?.addOperation {
+                AppSyncLog.debug("DS: =============== Perform Sync Timer =============== ")
+                self.performDeltaSync()
+            }
+        }
+    }
+    
+    // for first call, always try to fetch and return from cache.
+    func runBaseQueryFromCache() {
+        if let baseQuery = baseQuery {
+            let dispatchGroup = DispatchGroup()
+            dispatchGroup.enter()
+            AppSyncLog.info("DS: Running base query from cache.")
+            appsyncClient?.fetch(query: baseQuery, cachePolicy: CachePolicy.returnCacheDataDontFetch, resultHandler: {[weak self] (result, error) in
+                self?.baseQueryHandler?(result, error)
+                dispatchGroup.leave()
+            })
+            dispatchGroup.wait()
+        }
+    }
+    
+    // Each step represents whether to proceed to next step.
+    func runBaseQuery() -> Bool {
+        var success: Bool = true
+        if let baseQuery = baseQuery {
+            AppSyncLog.info("DS: Running Base Query Now")
+            let dispatchGroup = DispatchGroup()
+            dispatchGroup.enter()
+            let networkFetchTime = Date()
+            AppSyncLog.info("DS: Running base query from network.")
+            appsyncClient?.fetch(query: baseQuery, cachePolicy: .fetchIgnoringCacheData, resultHandler: {[weak self] (result, error) in
+                // call customer if successful or error
+                // return false to parent if failed
+                if error == nil {
+                    self?.baseQueryHandler?(result, error)
+                    success = true
+                } else if error != nil && result != nil {
+                    self?.baseQueryHandler?(result, error)
+                    success = true
+                } else {
+                    self?.baseQueryHandler?(result, error)
+                    success = false
+                }
+                if (success) {
+                    AppSyncLog.debug("DS: Updating last sync time.")
+                    self?.updateLastSyncTimeInMemoryAndCache(date: networkFetchTime)
+                }
+                dispatchGroup.leave()
+            })
+            dispatchGroup.wait()
+            
+        }
+        return success
+    }
+    
+    
+    /// Runs the delta query based on the given criterias.
+    ///
+    /// - Returns: true if the operation is executed successfully.
+    func runDeltaQuery() -> Bool {
```

Comment:
Is the return value necessary? It never returns anything but `true`

Reply:
It is a miss. should return false on a failed response. Fixing.